src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -1225,6 +1323,37 @@ void CodegenNeuronCppVisitor::print_make_node_data() const {
     printer->pop_block();
 }
 
+void CodegenNeuronCppVisitor::print_thread_variables_structure(bool print_initializers) {
+    if (codegen_thread_variables.empty()) {
+        return;
+    }
+
+    printer->add_newline(2);
+    printer->fmt_push_block("struct {} ", thread_variables_struct());
+    printer->add_line("double * thread_data;");
+    printer->add_newline();
+
+    std::string simd_width = "1";
```

Comment:
IIUC, currently this is always 1 but in the future something we can parameterise based on the execution environment / target.